packages/jellyfish-api-core/__tests__/category/masternode/createMasternode.test.ts

```diff
@@ -0,0 +1,64 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import { MasternodeState } from '../../../src/category/masternode'
+
+describe('Masternode', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should create a masternode transaction', async () => {
+    const masternodesBefore = await client.masternode.listMasternodes()
+    const masternodesLengthBefore = Object.keys(masternodesBefore).length
+
+    const address = await client.wallet.getNewAddress()
+    const masternode = await client.masternode.createMasternode(address)
+
+    await container.generate(1)
+
+    const masternodesAfter = await client.masternode.listMasternodes()
+    const masternodesLengthAfter = Object.keys(masternodesAfter).length
+
+    expect(masternodesLengthAfter).toStrictEqual(masternodesLengthBefore + 1)
+    expect(typeof masternode).toStrictEqual('string')
+
+    for (const masternode in masternodesAfter) {
+      const createdMasternode = masternodesAfter[masternode]
+      if (createdMasternode.ownerAuthAddress === address) {
```

Comment:
Should not be optional. I think it sorted desc chronologically, the first array component should have this particular address.

Reply:
Wrong. If you run the test you will know it is not the first one

Reply:
Yes, @jingyi2811  is correct. I just check it again.

Reply:
https://github.com/DeFiCh/jellyfish/pull/372/files#r650638933
https://github.com/DeFiCh/jellyfish/pull/372/files#r650639057

Shall we have this? Assertions in this for loop do not run if none of the listed MN has matching address and test will still pass.